cmd/talosctl/cmd/talos/edit.go

```diff
@@ -144,6 +139,21 @@ func editFn(c *client.Client) func(context.Context, client.ResourceResponse) err
 	}
 }
 
+func stripEditingComment(in []byte) []byte {
```

Comment:
we strip comments which start with `# ` from the header, as these comments are injected by the edit process and if we don't strip them, they will end up in the persisted machine config (as config is now preserved byte by byte)